### DISCUSSION TITLE
[Question][Hw1] TypeError in pad_oracle() and dec_oracle()

### DISCUSSION THREAD

**Discussion by KAIST-JongchanPark** [prosyslab-classroom/cs348-information-security#486] [Q&A]
In oracle_python_v1_2.py, there are 2 functions that use sock.sendall().

When I try to access the oracle, below "sock.sendall()" of "pad_oracle()" throws TypeError.
```
sock.sendall("pad_oracle," + str1 + "," + str2 + "\n")
```

I think it should be changed into,

```
sock.sendall(("pad_oracle," + str1 + "," + str2 + "\n").encode())
```

, and dec_oracle() as well, because sendall() can send only bytes, not str.

Could you fix this error?

**Comment by 0xXcrypted** [comment]
Same situation to me.
When running on python ( > 3) calling the function pad_oracle(c0, c1) gives me error in the socket.

One solution i found was to use python 2.7 though...

However, the document recommends to use 3.6.

**Comment by KAIST-JongchanPark** [comment]
Assignment is fixed now. Thanks.
